src/main/scala/org/broadinstitute/hail/variant/Genotype.scala

```diff
@@ -201,6 +201,13 @@ class Genotype(private val _gt: Int,
       b.append(pl.map(_.mkString(",")).getOrElse("."))
     else
       b.append(dosage.map(_.mkString(",")).getOrElse("."))
+
+    val regex = """(?::\.)+$""".r
+    val regexMatch = regex.findFirstMatchIn(b)
+    if (regexMatch.isDefined) {
+      b.delete(regexMatch.get.start, regexMatch.get.end)
+    }
```

Comment:
You can write this `regexMatch.foreach { m => b.delete(m.start, m.end) }`.  We use Option as a NullPointerException-safe alternative to pointers.  For this reason, we don't use `get` since it can throw `NoSuchElementException`.  Instead, use the monadic operators or pattern matching.  

This is a great change, I'll merge it once you address this.  I noticed some additional contextual improvements: https://github.com/broadinstitute/hail/issues/631.  Don't feel obligated to address them if you'd rather attack something else.